### PR TITLE
Use serial number from OTP and disable USB MSC

### PR DESCRIPTION
**Pull request by MistrKernnunos**
solves #2558 and #2542
BFW-2945

### CODE REVIEW

File: src/common/otp.c [hunk 1]
```diff
@@ -13,3 +13,7 @@ const char *otp_get_mac_address_str() {
     }
     return mac_address_str;
 }
+
+const char *otp_get_serial_number() {
+    return (const char *)(OTP_SERIAL_NUMBER_ADDR);
```
Comment (vorner) [inline src/common/otp.c:18]: I'm not sure this is valid. Isn't this a cast from volatile to non-volatile?

**Review by vorner — COMMENTED** [review]
I have a concern about a possible UB there. Can we get away without the unvolatilement of the pointer?

File: src/common/otp.c (hunk 1 above)
Comment (MistrKernnunos) [reply src/common/otp.c]: fixed

File: src/common/otp.c [hunk 2]
```diff
@@ -13,3 +13,16 @@ const char *otp_get_mac_address_str() {
     }
     return mac_address_str;
 }
+
+const char *otp_get_serial_number() {
+    static char serial_number_str[16] = { 0 };
```
Comment (lukash) [inline src/common/otp.c:18]: Shouldn't you make the size be `OTP_SERIAL_NUMBER_SIZE`? You have a mismatch in the size of the array and the upper bound of the `i` when you write to it below. I assume they're equal and chances of `OTP_SERIAL_NUMBER_SIZE` changing are low, but anyway.

**Review by vorner — APPROVED** [review]

**Pull request merged**